### PR TITLE
Convert UCAI to a namespace package

### PR DESCRIPTION
**Pull request by BenWilson2**
**PR Checklist**

- [X] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

This PR converts the ucai package to the unitycatalog.ai namespace, creating a namespace package that can have a unified interface that is shared with the official python SDK.

### CODE REVIEW

File: .github/workflows/ucai-integration-tests.yml [hunk 1]
```diff
@@ -56,11 +61,16 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install dependencies
+      - name: Install Core Library
+        run: |
+          pip install core/.
+      - name: Set PYTHONPATH for Core Library
+        run: |
+          echo "PYTHONPATH=$(pwd)/core/src:\$PYTHONPATH" >> $GITHUB_ENV
```
Comment (BenWilson2) [inline .github/workflows/ucai-integration-tests.yml:69]: This is to expose availability of the `test_utils` directory, which is not included in in the installation of the core package.

File: unitycatalog-ai/core/pyproject.toml [hunk 2]
```diff
@@ -55,11 +55,13 @@ include = [
...
 exclude = [
-  "src/ucai/test_utils/*"
+  "src/unitycatalog/ai/test_utils/*"
 ]
 
 [tool.hatch.build.targets.wheel]
-packages = ["src/ucai"]
+sources = ["src"]
+only-include = ["src/unitycatalog/ai/core"]
+packages = ["src/unitycatalog"]
```
Comment (BenWilson2) [inline unitycatalog-ai/core/pyproject.toml:64]: This is the "new" (as of Python 3.4) way of constructing namespace packages according to https://packaging.python.org/en/latest/guides/packaging-namespace-packages/ where the omission of __init__.py at the shared namespace(s) level is the recommended approach. 
We are intentionally not doing the legacy approach (https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#legacy-namespace-packages) that involves specifying the `pkg_resources` package from setuptools due to its marked deprecation and the higher level of complexity involved / higher chances of accidentally releasing a plugin module that will effectively break all other packages in the shared namespace.

File: unitycatalog-ai/integrations/llama_index/pyproject.toml [hunk 3]
```diff
@@ -28,11 +28,13 @@ build-backend = "hatchling.build"
...
 include = [
-  "src/ucai_llamaindex/*"
+  "src/unitycatalog/ai/llama_index/*"
 ]
 
 [tool.hatch.build.targets.wheel]
-packages = ["src/ucai_llamaindex"]
+sources = ["src"]
+only-include = ["src/unitycatalog/ai/llama_index"]
+packages = ["src/unitycatalog"]
```
Comment (serena-ruan) [inline unitycatalog-ai/integrations/llama_index/pyproject.toml:37]: Seems like this is not needed https://hatch.pypa.io/1.12/config/build/#packages

**Review by serena-ruan — APPROVED** [review]
Mostly LGTM! This PR is validated by https://github.com/databricks-eng/ai-oss-integration-tests-runner/pull/8 as well :)

File: unitycatalog-ai/integrations/llama_index/pyproject.toml (hunk 3 above)
Comment (BenWilson2) [reply unitycatalog-ai/integrations/llama_index/pyproject.toml]: Good catch! Removed redundant option :)

File: .github/workflows/ucai-integration-tests.yml [hunk 4]
```diff
@@ -56,11 +61,16 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install dependencies
+      - name: Install Core Library
+        run: |
+          pip install core/.
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply .github/workflows/ucai-integration-tests.yml]: Good call :) No need to have the distinct stages here

**Review by dennyglee — APPROVED** [review]
LGTM!

**Pull request merged**